vcenter_operator/vcenter_sso.py

```diff
@@ -0,0 +1,205 @@
+            )
+            raise SSOSkippedError()
+
+    def list_service_users(self, host, search_string, limit=10000):
+        """List service-users in the vCenter via SSO instance"""
+        api = self._get_api_instance(host)
+
+        try:
+            principal_discovery_service = api.principalDiscoveryService
+            criteria = pyVmomi.sso.AdminPrincipalDiscoveryServiceSearchCriteria(
+                searchString=search_string, domain=self.domain
+            )
+            users = principal_discovery_service.FindUsers(criteria=criteria, limit=limit)
+            user_names = [user.id.name for user in users]
+        except Exception as e:
+            LOG.error("Error listing service-users: %s", e)
+            del self.sso_admin_instances[host]
+            raise SSOSkippedError()
+
+        return user_names
+    
+    def check_users_in_group(self, host, search_string, limit=10000):
+        """Checks if service-user is in the Administrators group in the vCenter via SSO instance"""
+        api = self._get_api_instance(host)
+
+        try:
+            principal_discovery_service = api.principalDiscoveryService
+            criteria = pyVmomi.sso.AdminPrincipalDiscoveryServiceSearchCriteria(
+                searchString="", domain=self.domain
+            )
+            groups = principal_discovery_service.FindGroups(criteria=criteria, limit=limit)
+            for group in groups:
+                if group.id.name != "Administrators":
+                    continue
+                users = principal_discovery_service.FindUsersInGroup(searchString=search_string,groupId=group.id, limit=limit)
+                for user in users:
+                    if user.id.name == search_string:
+                        LOG.debug("User %s is in the Administrators group in vCenter %s", search_string, host)
+                        return True
+                return False
+        except Exception as e:
+            LOG.error("Error checking service-users in Administrator group: %s", e)
+            del self.sso_admin_instances[host]
+            raise SSOSkippedError()
+
+    def create_service_user(self, host, username, password, service):
+        """Create a service-user in the vCenter via SSO instance"""
+        api = self._get_api_instance(host)
+        description = "Service-user for service " + service
+
+        try:
+            principal_management_service = api.principalManagementService
+            details = pyVmomi.sso.AdminPersonDetails(description=description)
+
+            if self.dry_run:
+                LOG.debug(f"Dry-run: Would have created service-user in vcenter {host}")
+                return
+            resp = principal_management_service.CreateLocalPersonUser(
+                userName=username, userDetails=details, password=password
+            )
+            if resp and resp.name and resp.name != "":
+                LOG.info("Successfully created service-user in vCenter %s.", host)
```

Comment:
What about the error-case here? We don't log it, because it didn't raise an exception.